fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewPartitionFunctionChecker.java

```diff
@@ -0,0 +1,134 @@
+            return false;
+        }
+
+        if (!(fnExpr.getChild(0) instanceof StringLiteral)) {
+            return false;
+        }
+        String fmt = ((StringLiteral) fnExpr.getChild(0)).getValue();
+        if (fmt.equalsIgnoreCase("week")) {
+            throw new SemanticException("The function date_trunc used by the materialized view for partition" +
+                    " does not support week formatting", expr.getPos());
+        }
+
+        Expr child1 = fnExpr.getChild(1);
+        if (child1 instanceof SlotRef) {
+            SlotRef slotRef = (SlotRef) child1;
+            PrimitiveType primitiveType = slotRef.getType().getPrimitiveType();
+            // must check slotRef type, because function analyze don't check it.
+            return primitiveType == PrimitiveType.DATETIME || primitiveType == PrimitiveType.DATE;
+        } else if (child1 instanceof FunctionCallExpr) {
+            // date_trunc('hour', time_slice(dt, 'minute'))
+            FunctionCallExpr funcExpr = (FunctionCallExpr) child1;
+            if (funcExpr.getFnName().getFunction().equalsIgnoreCase(FunctionSet.TIME_SLICE)) {
+                // TODO: support more functions which do not affect the mv's final partition.
+                Expr child0 = funcExpr.getChild(0);
+                if (!(child0 instanceof SlotRef)) {
+                    throw new SemanticException("1th child of the function time_slice used by the materialized view " +
+                            "only supports slot ref", expr.getPos());
+                }
+                // Check time_slice's input type valid
+                SlotRef slotRef = (SlotRef)  child0;
+                PrimitiveType primitiveType = slotRef.getType().getPrimitiveType();
+                if (primitiveType != PrimitiveType.DATETIME && primitiveType != PrimitiveType.DATE) {
+                    return false;
+                }
+                // NOTE: Ensure time_slice's time unit is less than partition's time unit which
+                // will not affect the mv's final partition.
+                // TODO: Remove this if we can support more elastic auto-partitioned for MV.
+                String timeSliceFmt = ((StringLiteral) funcExpr.getChild(2)).getValue();
+                if (TIME_MAP.containsKey(timeSliceFmt) && TIME_MAP.containsKey(fmt) &&
+                        TIME_MAP.get(timeSliceFmt) < TIME_MAP.get(fmt)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public static boolean checkTimeSlice(Expr expr) {
+        if (!(expr instanceof FunctionCallExpr)) {
+            return false;
+        }
+        FunctionCallExpr fnExpr = (FunctionCallExpr) expr;
+        String fnNameString = fnExpr.getFnName().getFunction();
+        if (!fnNameString.equalsIgnoreCase(FunctionSet.TIME_SLICE)) {
+            return false;
+
+        }
+        List<Expr> paramExprs = fnExpr.getParams().exprs();
+        if (paramExprs.size() != 4) {
```

Comment:
better check the forth paramExprs must be floor.

Reply:
fix